### PR TITLE
check for vxlan mapping before removing vlan

### PR DESCRIPTION
**Pull request by ITJamie**
issue: https://github.com/sonic-net/sonic-utilities/issues/2381

#### What I did
check for vxlan mapping before removing vlan

#### How I did it

added a check similar to vlan member above

#### How to verify it

attempt to delete a vlan that has a vxlan mapping in place

### CODE REVIEW

File: config/vlan.py [hunk 1]
```diff
@@ -60,6 +60,11 @@ def del_vlan(db, vid):
     
     if keys:
         ctx.fail("VLAN ID {} can not be removed. First remove all members assigned to this VLAN.".format(vid))
+
+    vxlan_table = db.cfgdb.get_table('VXLAN_TUNNEL_MAP')
```
Comment (prsunny) [inline config/vlan.py:64]: why do we have to set this restriction? will there be any ASIC failures?

Comment (ITJamie) [reply config/vlan.py:64]: it is leaving behind vxtunnel maps in the configdb to vlans that have been removed. its messy

**Review by dgsudharsan — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@ITJamie  Can you please resolve conflicts?

**Comment by ITJamie** [comment]
conflicts resolved

**Comment by ITJamie** [comment]
tests are passing now

**Review by prsunny — APPROVED** [review]

**Comment by yxieca** [comment]
@ITJamie please raise separate PR for 202205. This PR cannot be cherry-picked cleanly.

**Comment by dgsudharsan** [comment]
@ITJamie  Can you please raise a PR for 202205?

**Pull request merged**